src/client/dfs/dfs.c

```diff
@@ -2370,8 +2370,7 @@ dfs_open(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 		ret = daos_tx_close(th,  NULL);
 		if (ret) {
 			D_ERROR("daos_tx_close() failed (%d)\n", ret);
-			if (rc == 0)
-				rc = daos_der2errno(ret);
+			rc = daos_der2errno(ret);
```

Comment:
I don't think this one is right either.

Reply:
rc will never be 0 when the code gets here, normally came here via some error like: D_GOTO(out, rc = EINVAL);